cloud/os_compute_facts.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python
+
+# Copyright (c) 2014 Hewlett-Packard Development Company, L.P.
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+try:
+    import shade
+    from shade import meta
+except ImportError:
+    print("failed=True msg='shade is required for this module'")
+
+DOCUMENTATION = '''
+---
+module: os_compute_facts
+short_description: Retrieve facts about a compute instance
+extends_documentation_fragment: openstack
+description:
+   - Retrieve facts about a compute instance from OpenStack
+options:
+   name:
+     description:
+        - Name of the instance
+     default: None
+   id:
```

Comment:
Thanks. A future revision of this PR will try to use 'server' to contain either a server instance name or ID.